File: src/unit-test-data/Data.Tests/Tests.DataTest/String/Test.String.cs
using Xunit;

namespace PrimeFuncPack.UnitTest.Data.Tests;

partial class TestDataTest
{
    [Fact]
    public void EmptyString_ExpectEmptyString()
    {
        var actual = TestData.EmptyString;
        Assert.Empty(actual);
    }

    [Fact]
    public void WhiteSpaceString_ExpectWhiteSpace()
    {
        var actual = TestData.WhiteSpaceString;

        var expected = new string(' ', 1);
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void TwoWhiteSpacesString_ExpectTwoWhiteSpacesString()
    {
        var actual = TestData.TwoWhiteSpacesString;

        var expected = new string(' ', 2);
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void ThreeWhiteSpacesString_ExpectThreeWhiteSpacesString()
    {
        var actual = TestData.ThreeWhiteSpacesString;

        var expected = new string(' ', 3);
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void TabString_ExpectTabString()
    {
        var actual = TestData.TabString;

        var expected = new string('\t', 1);
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void TwoTabsString_ExpectTwoTabsString()
    {
        var actual = TestData.TwoTabsString;

        var expected = new string('\t', 2);
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void MixedWhiteSpacesString_ExpectTheMixedWhiteSpacesString()
    {
        var actual = TestData.MixedWhiteSpacesString;

        var expected = new string(new[] { '\t', '\t', ' ' });
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void SomeString_ExpectTextIsSomeString()
    {
        var actual = TestData.SomeString;

        var expected = "Some String";
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void LowerSomeString_ExpectTextIsResultOfSomeStringToLower()
    {
        var actual = TestData.LowerSomeString;

        var expected = TestData.SomeString.ToLowerInvariant();
        Assert.Equal(expected, actual, true);
    }

    [Fact]
    public void UpperSomeString_ExpectTextIsResultOfSomeStringToUpper()
    {
        var actual = TestData.UpperSomeString;

        var expected = TestData.SomeString.ToUpperInvariant();
        Assert.Equal(expected, actual, true);
    }

    [Fact]
    public void AnotherString_ExpectTextIsSomeString()
    {
        var actual = TestData.AnotherString;

        var expected = "Another string";
        Assert.Equal(expected, actual);
    }

    [Fact]
    public void LowerAnotherString_ExpectTextIsResultOfSomeStringToLower()
    {
        var actual = TestData.LowerAnotherString;

        var expected = TestData.AnotherString.ToLowerInvariant();
        Assert.Equal(expected, actual, true);
    }

    [Fact]
    public void UpperAnotherString_ExpectTextIsResultOfSomeStringToUpper()
    {
        var actual = TestData.UpperAnotherString;

        var expected = TestData.AnotherString.ToUpperInvariant();
        Assert.Equal(expected, actual, true);
    }
}

File: src/unit-test-data/UnitTest.Data.Tests/Tests.DataTest/Test.String.fs
namespace PrimeFuncPack.UnitTest.Data.Tests.TestData

open Xunit
open PrimeFuncPack.UnitTest
open TestHelper

module StringTest =

    let private toString chars =
        new string([| for c in chars -> c |])

    let private createString (char: char) count =
        new string([| for _ = 1 to count do char |])

    let private toLowerInvariant (text: string) =
        text.ToLowerInvariant()

    let private toUpperInvariant (text: string) =
        text.ToUpperInvariant()

    [<Fact>]
    let ``TestData.EmptyString sould be the empty string``() =
        TestData.EmptyString |> shouldBeEmpty

    [<Fact>]
    let ``TestData.WhiteSpaceString sould be the white space string``() =
        let expected = createString ' ' 1
        TestData.WhiteSpaceString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.TwoWhiteSpacesString sould be the two white spaces string``() =
        let expected = createString ' ' 2
        TestData.TwoWhiteSpacesString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.ThreeWhiteSpacesString sould be the three white spaces string``() =
        let expected = createString ' ' 3
        TestData.ThreeWhiteSpacesString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.TabString sould be the tab string``() =
        let expected = createString '\t' 1
        TestData.TabString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.TwoTabsString sould be two the tabs string``() =
        let expected = createString '\t' 2
        TestData.TwoTabsString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.MixedWhiteSpacesString sould be the mixed white spaces string``() =
        let expected = [| '\t'; '\t'; ' ' |] |> toString
        TestData.MixedWhiteSpacesString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.SomeString sould be the some string``() =
        TestData.SomeString |> shouldBeEqualString "Some String"

    [<Fact>]
    let ``TestData.LowerSomeString sould be the lower some string``() =
        let expected = TestData.SomeString |> toLowerInvariant
        TestData.LowerSomeString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.UpperSomeString sould be the upper some string``() =
        let expected = TestData.SomeString |> toUpperInvariant
        TestData.UpperSomeString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.AnotherString sould be the other string``() =
        TestData.AnotherString |> shouldBeEqualString "Another string"

    [<Fact>]
    let ``TestData.LowerAnotherString sould be the lower other string``() =
        let expected = TestData.AnotherString |> toLowerInvariant
        TestData.LowerAnotherString |> shouldBeEqualString expected

    [<Fact>]
    let ``TestData.UpperAnotherString sould be the upper other string``() =
        let expected = TestData.AnotherString |> toUpperInvariant
        TestData.UpperAnotherString |> shouldBeEqualString expected